cmd/gh/main.go

```diff
@@ -186,7 +186,9 @@ func printError(out io.Writer, err error, cmd *cobra.Command, debug bool) {
 }
 
 func shouldCheckForUpdate() bool {
-	return updaterEnabled != "" && !isCompletionCommand() && utils.IsTerminal(os.Stderr)
+	isCIEnvironment := (os.Getenv("CI") != "") || (os.Getenv("BUILD_NUMBER") != "") || (os.Getenv("RUN_ID") != "")
+	isGlobalDisabled := os.Getenv("GH_NO_UPDATE_NOTIFIER") == "disabled"
```

Comment:
Could this just check for `GH_NO_UPDATE_NOTIFIER` being set to any non-blank string, like the above checks?

The presence of "NO" in this variable implies that if the variable is set (to any value), update notifier will be disabled.